### DISCUSSION TITLE
Incorrect highlighting of  parameter in function body

### DISCUSSION THREAD

**Discussion by LDT9510** [nvim-treesitter/nvim-treesitter#5196] [Q&A] [answered]
![image](https://github.com/tree-sitter/tree-sitter/assets/31300566/4d98b30b-014c-4fc8-bace-3390dff7e628)

Hello, as you can see in the image the parameter is correctly highlighted in the function definition, but is treated as a variable in the body. Idk if this is related directly with TreeSitter, I'm very new to the nvim world. I appreciate any help or guidance towards a solution. Running `:TSHighlightCapturesUnderCursor` in the parameter shows:
```
# Treesitter
* **@variable**
* **@parameter**
```
And in the function body:
```
# Treesitter
* **@variable**
```
Dunno know if that helps. Thx.

Edit: Apparently this happens for Rust too, but is fine for Typescript. Any ideas?

**Comment by amaanq** [comment, marked as answer]
that needs lsp semantic tokens
